test/unittest/blas1/blas1_rotg_test.cpp

```diff
@@ -83,8 +96,29 @@ void run_test(const combination_t<scalar_t> combi) {
   ASSERT_TRUE(utils::almost_equal(s, s_ref));
 }
 
+template <typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  std::string alloc;
+  api_type api;
+  scalar_t a_input;
+  scalar_t b_input;
+  scalar_t c_input = 1;
+  scalar_t s_input = 1;
+
+  std::tie(alloc, api, a_input, b_input) = combi;
+
+  if (alloc == "usm") {  // usm alloc
+#ifdef SB_ENABLE_USM
+    run_test<scalar_t, helper::AllocType::usm>(combi);
+#endif
+  } else {  // buffer alloc
+    run_test<scalar_t, helper::AllocType::buffer>(combi);
+  }
+}
```

Comment:
What do you think about moving this code to a common place (util_test of something)? I have seen this same block of code for all tests.

Reply:
I couldn't move this code to a more common place since the `combi` object for each test doesn't necessarily need to be the same. That's the reason I decided to keep this code in every test separately to keep the code consistent.